### ISSUE TITLE
Scaled Z axis with MANUAL_BED_LEVELING and MESH_BED_LEVELING

### ISSUE THREAD

**Issue by oscargilfc** [MarlinFirmware/Marlin#2739]
Additional to homing problem #1682
When MESH_BED_LEVELING is used Z axis is scaled and it moves about 1,1 times more when going up, going down there appears there is no problem. I made tests and I think is software error, if I comment the next lines z axis works properly. 

```
#define MANUAL_BED_LEVELING  // Add display menu option for bed leveling.
#define MESH_BED_LEVELING 
```

I have not tested with AUTO_BED_LEVELING because I don't have the required hardware. I tried with mesh sizes of 3x3 4x4 and 5x5

In short distances is not noticeable, but if I move it 100 mm it goes more than 110. It sounds like it have very short pauses. In 2 occasions my board reset, I don't know if it was another thing but that only had happened to me with those options. 

 I first thought it was my stepping configuration, but when I disable those It works good.

**Comment by epatel** [comment]
Hi, great to hear you trying the `MESH_BED_LEVELING`, and sorry you having problems with it.
You testing just moving it straight up? like `G1 Z100`? Do you do this just after homing, `G28`?
You might have a version where `G28` does not go to `Z=0` but rather leave the nozzle slightly above the bed. The movements should be correct, it just starts off a little bit above the bed.
I added some code to fix this in this pull request https://github.com/MarlinFirmware/MarlinDev/pull/199 (for this issue https://github.com/MarlinFirmware/MarlinDev/issues/162) but it doesn't seem to have been merged.
So, I recommend doing a `G1 Z0` just after the `G28`.

**Comment by oscargilfc** [comment]
Hello, I think is not the offset, I measure the displacement with a caliper. I fixed the Z=0 after homing and still have that issue. I have tested moving just after G28, I will perform up and down tests and see if the same happens after going down, I've actually don't remember if tried that, may be y my z=0 fix is wrong, I will try with your code fix.

**Comment by epatel** [comment]
What kind of printer do you have? I have only tested on "normal" none delta

**Comment by Roxy-3D** [comment]
> What kind of printer do you have? I have only tested on "normal" none delta

@epatel Have you looked at how Delta's do Auto Bed Leveling?  It seems it is really just an extension or automatic way of doing what you came up with.   When you have some spare time and feel like doing it, can you compare your stuff with what Delta's do?  I think there might be an opportunity to combine the two.

**Comment by thinkyhead** [comment]
How is this looking? If there's still an issue, try using the `DEBUG_LEVELING_FEATURE` with `M115 S32` and checking the output for any weird results.

**Comment by thinkyhead** [comment]
Is this still an issue with the current `RCBugFix` code or did we manage to patch it?

**Comment by oscargilfc** [comment]
I will test it and post the results.

**Comment by jbrazio** [comment]
Thank you for your interest making Marlin better and reporting this issue but this topic has been open for a long period of time without any further development. Marlin has been under heavy development for the past couple of months and moving to it's last mile to finish the RC cycle and release **Marlin v1.1.0**. We suggest you to try out the latest [RCBugfix](https://github.com/MarlinFirmware/Marlin/archive/RCBugFix.zip) branch and reopening this issue if required.
